drivers/azure/destroy.go

```diff
@@ -3,3 +3,209 @@
 // that can be found in the LICENSE file.
 
 package azure
+
+import (
+	"context"
+	"fmt"
+	"github.com/drone/autoscaler"
+	"github.com/rs/zerolog/log"
+	disks "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+)
+
+func (p *provider) getDiskClient() (disks.DisksClient, error) {
+	disksClient := disks.NewDisksClient(p.subscriptionId)
+	auth, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		return disksClient, err
+	}
+	disksClient.Authorizer = auth
+	return disksClient, nil
+}
+
+func (p *provider) deleteDisks(ctx context.Context, instance *autoscaler.Instance) error {
+	logger := log.Ctx(ctx).With().
+		Str("Azure Destroy:", instance.ID).
+		Str("id", instance.ID).
+		Str("ip", instance.Address).
+		Str("name", instance.Name).
+		Str("zone", instance.Region).
+		Logger()
+
+	diskClient, err := p.getDiskClient()
+	if err != nil {
+		logger.Error().
+			Err(err).
+			Msg("cannot terminate instance")
+		return err
+	}
+
+	future, err := diskClient.Delete(ctx, p.resourceGroup, instance.ID + "-osdisk")
+	err = future.WaitForCompletionRef(ctx, diskClient.Client)
+	logger.Debug().Msg("waitforcompletionref")
+	if err != nil {
+		logger.Error().
+			Err(err).
+			Msg("cannot terminate instance")
+		return err
+	}
+	
+	if err != nil {
+		logger.Error().
+			Err(err).
+			Msg("cannot terminate instance")
+		return err
+	}
```

Comment:
this error is already handled at line 46

Reply:
Yep will remove